insights/specs/default.py

```diff
@@ -428,23 +430,41 @@ def semid(broker):
-    oc_get_configmap = simple_command("/usr/bin/oc get configmap -o yaml --all-namespaces", context=OpenShiftContext)
+
+    oc_get_nodes = simple_command("oc get nodes", inherit_env=["KUBECONFIG"])
+
+    @datasource(oc_get_nodes, c_hostname)
+    def is_openshift_master(broker):
+        """ Determine if this is the openshift master node. """
+        raw_nodes = broker[DefaultSpecs.oc_get_nodes].content
+        nodes = parse_fixed_table(raw_nodes)
+        if len(nodes) == 1:
+            return
+
+        fqdn = broker[c_hostname].fqdn.lower()
+        for node in nodes:
+            if node["NAME"] == fqdn and "master" in node["ROLES"].lower():
+                return
+
+        raise dr.SkipComponent()
+
+    oc_get_bc = simple_command("oc get bc -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_build = simple_command("oc get build -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_dc = simple_command("oc get dc -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_egressnetworkpolicy = simple_command("oc get egressnetworkpolicy -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_endpoints = simple_command("oc get endpoints -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_event = simple_command("oc get event -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_node = simple_command("oc get nodes -o yaml", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_pod = simple_command("oc get pod -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_project = simple_command("oc get project -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_pv = simple_command("oc get pv -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_pvc = simple_command("oc get pvc -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_rc = simple_command("oc get rc -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_role = simple_command("oc get role -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_rolebinding = simple_command("oc get rolebinding -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_route = simple_command("oc get route -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_service = simple_command("oc get service -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
+    oc_get_configmap = simple_command("oc get configmap -o yaml --all-namespaces", deps=[is_openshift_master], inherit_env=["KUBECONFIG"])
```

Comment:
@csams could you create an `oc_simple_command` derived from `simple_command` to encapsulate all of the common stuff here and just pass args for spec specific stuff?  Can we create a hierarchy like that in specs?

Reply:
This same question applies to commands like `ls` as well.

Reply:
I think so. We'd have to define it within the `DefaultSpecs` class so we can make sure they all have the same deps. I'll look into it.